### ISSUE TITLE
:protocol missing

### ISSUE THREAD

**Issue by malcolmsparks** [ring-clojure/ring-mock#10]
Mock request does not generate :protocol as per Ring [SPEC](https://github.com/ring-clojure/ring/blob/master/SPEC)

```
:protocol
  (Required, String)
  The protocol the request was made with, e.g. "HTTP/1.1".
```

and fails to conform to the following spec:

```
(s/def :ring/request
  (s/keys :req-un [:ring.request/server-port
                   :ring.request/server-name
                   :ring.request/remote-addr
                   :ring.request/uri
                   :ring.request/scheme
                   :ring.request/protocol
                   :ring.request/headers
                   :ring.request/request-method]
          :opt-un [:ring.request/query-string
                   :ring.request/body]))
```

from [ring-spec](https://github.com/ring-clojure/ring-spec/blob/master/src/ring/core/spec.clj#L115)

**Comment by malcolmsparks** [comment]
I found this when playing around with ring-spec - which is great btw. Is ring-mock deprecated now or does it just need a fix? Happy to send a PR.

**Comment by malcolmsparks** [comment]
Also, :query-string is generated as nil, which fails the spec.

```clojure
(s/explain-str :ring/request (assoc (request :get "/") :protocol "HTTP/1.1"))
=>
In: [:query-string] val: nil fails spec: :ring.request/query-string at: [:query-string] predicate: :clojure.spec/unknown
```

**Comment by weavejester** [comment]
Needs a fix, I think! Ring-Mock is still useful for creating non-generative request maps. The `:query-string` key should probably be missing if it's not set.

**Comment by malcolmsparks** [comment]
I don't know how you feel about the ring-mock project now, given that equivalent functionality could eventually form part of the ring-spec project (via generators). Personally, I like ring-mock and think it's worth keeping up-to-date. Feel free to delegate to me if you like, but I think the fixes are fairly trivial.

**Comment by weavejester** [comment]
As far as I know, generators aren't designed for creating specific data structures, so Ring-Mock is still useful for creating non-generative request maps.

**Comment by jcf** [comment]
This should be fixed now that #12 has been merged.
